### PR TITLE
feat: change DataView Grouping `compileAccumulatorLoop` for CSP safe 

### PR DESCRIPTION
**Pull request by JesperJakobsenCIM**

### CODE REVIEW

File: src/slick.dataview.ts
Comment (ghiscoding) [inline src/slick.dataview.ts:434]: small lint nitpick, can you add space after curly brace and also add a space after bracket on the if condition.

File: src/slick.dataview.ts
Comment (ghiscoding) [inline src/slick.dataview.ts:1065]: can you remove the extra line carriage above

**Review by ghiscoding — CHANGES_REQUESTED** [review]

**Comment by ghiscoding** [comment]
Apart from some small Linting issues, the PR looks quite simple. Have you check logged perf? Just asking again in case there's no difference in perf, if so only 1 approach is enough. Remember that we kept the other separate because there was a slight perf decrease

**Comment by JesperJakobsenCIM** [comment]
50 rows
image
Small test, first one is with CSP and second test is without CSP (old) method

By general the new method actually seems faster, surprisingly me a little and its even with quite a margin of performance

This is old method first and new method next with 5000 rows instead of 50
image

**Comment by ghiscoding** [comment]
it seems very similar and small diff, I would probably vote for keeping only the new one, what do you think?

**Comment by ghiscoding** [comment]
@JesperJakobsenCIM so have you had a chance to compare if changing to a regular for loop makes a difference in performance? Do you think we should keep both function or merged into a single function? I'm waiting for your feedback before merging the PR. Thanks

**Comment by JesperJakobsenCIM** [comment]
the 50.000 example is with a for loop, and it seems to get faster since 5.000 rows (without for loop) has the same speed as 50.000 rows with for loop

**Comment by ghiscoding** [comment]
So does that mean that both functions are nearly the same speed? If so, can we keep only 1 function (the newest and CSP safe).

**Comment by JesperJakobsenCIM** [comment]
its actually marginally faster
image
50.000 rows test
first 5 rows is CSP Safe and last 5 rows is old function

**Comment by ghiscoding** [comment]
that is really awesome to see, faster is great 🚀 

So could refactor and keep only the CSP function then? I could merge it after and publish a new release, just waiting for this PR :)

**Comment by JesperJakobsenCIM** [comment]
I've removed the old method and kept new method with its name for clarity.

**Review by ghiscoding — APPROVED** [review]

**Comment by ghiscoding** [comment]
This is now available in the latest release [v5.7.0](https://github.com/6pac/SlickGrid/releases/tag/5.7.0), thanks again for your contributions. I think this was the last piece to be fully CSP Safe, thanks a lot for all your help and guidance on the subject 🎉

Happy Holidays 🎁

**Pull request merged**
